### PR TITLE
Only escape/remove octet if flagged in ACCM

### PR DESCRIPTION
**Pull request by samherrmann**
Please see commit messages for details.

### CODE REVIEW

File: accm.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package hdlc
+
+// ControlChar is an ASCII control character, which is a character between 0x00
+// and 0x20, inclusive.
+//
+// https://en.wikipedia.org/wiki/ASCII#Control_code_chart
+type ControlChar uint32
```
Comment (samherrmann) [inline accm.go:7]: There is one uncertainty I have about this type. Consider the following paragraph in [section 7.1 of RFC 1662](https://www.rfc-editor.org/rfc/rfc1662#section-7.1):

> Each end of the asynchronous link maintains two Async-Control-
      Character-Maps.  The receiving ACCM is 32 bits, but the sending
      ACCM may be up to 256 bits.  This results in four distinct ACCMs,
      two in each direction of the link.

I don't understand what the 256 bits would consist of.

Comment (zaninime) [reply accm.go:7]: I tried to gather more information but I didn't get any further unfortunately.

**Comment by zaninime** [comment]
Thanks a lot once more, and sorry for the long delay. I don't have a use case at hand for this atm, but as current functionality is not affected I have no problems in merging it.

**Comment by samherrmann** [comment]
@zaninime, thank YOU for merging! 🙂

**Comment by samherrmann** [comment]
@zaninime would you be ok with cutting a new release so that the example shows in the pkg.go.dev and the library is gettable by a clean version?

**Comment by zaninime** [comment]
Done! https://pkg.go.dev/github.com/zaninime/go-hdlc@v1.2.0 thanks for the hint.

**Pull request merged**